update website for a more modern homepage

agavra opened the pull request:
## Summary

Modernize the home page to gear up for an eventual 1.0 launch. The change trims down the content, reduces word count and adds some social proof to the home page. This also fixes the favicon rendering.

## Changes

<img width="1409" height="1327" alt="image" src="https://github.com/user-attachments/assets/250ffd34-e559-4c08-aefd-01a369431bf4" />


## Notes for Reviewers

LMK what you think 😄 

## Checklist

N/A

Thank you for the review! 🙏

criccomini reviewed (approved):
Nice! Was thinking we needed to do some more branding work. :D

Some notes, perhaps for follow-up:

- Use the favicon-style (no sparkle) logo everywhere. Sparkles really don't translate well to smaller formats.
- Add some text along with the icons for the upper right nav menu (maybe experiment with different styles or a hamburger menu)
- Highlight SlateDB features somewhere. We have a quite formidable list now (merge op, ttl, txn, multi-reader, range queries, db clones)
- Can we use the monospace font for the logo as well? (and eliminate bold in **DB*)
- Probably worth highlighting that it works with Python and Go, too. At a minimum, the hero terminal should have tabs for pip install and go get as well.
- Mind updating the social graph image, too? IIRC, it's still using the old logo/text.

agavra commented:
OK! I addressed most the feedback and updated the uploaded image - I went with text-only menu links and then icons in the footer. The page has a bit of dynamic styling (i.e. when you hover over features) to make it a little less static.
